jury/metrics/chrf/chrf_for_language_generation.py

```diff
@@ -0,0 +1,259 @@
+        output = sb_chrf.corpus_score(predictions, references)
+
+        return (
+            output.score / 100,
+            output.char_order,
+            output.word_order,
+            output.beta,
+        )
+
+    def _compute_single_pred_single_ref(
+        self,
+        predictions: LanguageGenerationInstance,
+        references: LanguageGenerationInstance,
+        reduce_fn: Callable = None,
+        char_order: int = None,
+        word_order: int = None,
+        beta: int = None,
+        lowercase: bool = False,
+        whitespace: bool = False,
+        eps_smoothing: bool = False,
+    ):
+        score, c_ord, w_ord, beta = self._compute_chrf_score(
+            predictions,
+            references,
+            char_order=char_order,
+            word_order=word_order,
+            beta=beta,
+            lowercase=lowercase,
+            whitespace=whitespace,
+            eps_smoothing=eps_smoothing,
+        )
+        return {"score": score, "char_order": c_ord, "word_order": w_ord, "beta": beta}
+
+    def _compute_single_pred_multi_ref(
+        self,
+        predictions: LanguageGenerationInstance,
+        references: LanguageGenerationInstance,
+        reduce_fn: Callable = None,
+        char_order: int = None,
+        word_order: int = None,
+        beta: int = None,
+        lowercase: bool = False,
+        whitespace: bool = False,
+        eps_smoothing: bool = False,
+    ):
+        # SacreBleu inherently supports multiple references.
+        return self._compute_single_pred_single_ref(
+            predictions=predictions,
+            references=references,
+            reduce_fn=reduce_fn,
+            char_order=char_order,
+            word_order=word_order,
+            beta=beta,
+            lowercase=lowercase,
+            whitespace=whitespace,
+            eps_smoothing=eps_smoothing,
+        )
+
+    def _compute_multi_pred_multi_ref(
+        self,
+        predictions: LanguageGenerationInstance,
+        references: LanguageGenerationInstance,
+        reduce_fn: Callable = None,
+        char_order: int = None,
+        word_order: int = None,
+        beta: int = None,
+        lowercase: bool = False,
+        whitespace: bool = False,
+        eps_smoothing: bool = False,
+    ):
+        scores = []
+        for preds, refs in zip(predictions, references):
+            pred_scores = []
+            for pred in preds:
+                score, _, _, _ = self._compute_chrf_score(
+                    pred,
+                    refs,
+                    char_order=char_order,
+                    word_order=word_order,
+                    beta=beta,
+                    lowercase=lowercase,
+                    whitespace=whitespace,
+                    eps_smoothing=eps_smoothing,
+                )
+                pred_scores.append(score)
+            scores.append(reduce_fn(pred_scores))
+
+        return {
+            "score": float(reduce_fn(scores)),
```

Comment:
Reverse Comment of TER PR: Why Reduce here instead of mean?

Reply:
Yes, this line is incorrect indeed, nice catch. :+1: